http-server/src/main/java/org/triplea/server/moderator/toolbox/api/key/InvalidKeyLockOut.java

```diff
@@ -44,14 +48,29 @@ public InvalidKeyLockOut(
    * @param request Request containing moderator api key as a header.
    */
   public boolean isLockedOut(final HttpServletRequest request) {
-    final boolean lockedOut = invalidKeyCache.getCount(request) >= maxFailsByIpAddress
-        || invalidKeyCache.totalSum() >= maxTotalFails;
-
-    log.warning("Request for API key validation by: " + request.getRemoteHost() + " is locked out");
-    return lockedOut;
+    if (invalidKeyCache.getCount(request) >= maxFailsByIpAddress
+        || invalidKeyCache.totalSum() >= maxTotalFails) {
+      log.warning("Request for API key validation by: " + request.getRemoteHost() + " is locked out");
+      return true;
+    }
+    return false;
   }
 
   public void recordInvalid(final HttpServletRequest request) {
     invalidKeyCache.increment(request);
   }
+
+  /**
+   * May only be used in non-prod, when called resets the lockout cache. This is to enable testing.
+   */
+  public synchronized void clearLockouts(final HttpServletRequest request) {
+    if (production) {
+      // if accessed in production, not allowed, immediately lock out.
```

Comment:
Bit noteworthy, if clearLockouts is accessed in prod, we do an immediate lockout of that IP.